app/src/org/commcare/activities/UpdateActivity.java

```diff
@@ -314,7 +316,16 @@ public void handleTaskCompletion(ResultAndError<AppInstallStatus> result) {
                 finishWithResult(RefreshToLatestBuildActivity.ALREADY_UP_TO_DATE);
             }
         } else {
-            reportFailureToNotifications(result.errorMessage);
+            switch(result.data) {
+                case BadSSLCertificate:
+                    reportFailureToNotifications(result.data, NotificationActionButtonInfo.ButtonAction.LAUNCH_DATE_SETTINGS);
+                    break;
+                default:
+                    //QUESTION: AppInstallStatus maps to MessageTags, shouldn't we be using that? i.e.
+                    //reportFailureToNotifications(result.data, NotificationActionButtonInfo.ButtonAction.NONE);
```

Comment:
Ok, I rearranged the code a bit so that the checks for which notification to show all happen in the reportFailureToNotification method. I also added a fallback case to create the NotificationMessage simply from the AppInstallStatus (no extra message).

The way the code is now, the UpdateFailedGeneral message with additional errorMessage is still shown if the errorMessage is not empty. The errorMessage usually comes from calling getMessage() on the caught exception.